client/state/data-layer/wpcom/concierge/schedules/appointments/book/index.js

```diff
@@ -0,0 +1,77 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { updateConciergeBookingStatus } from 'state/concierge/actions';
+import { errorNotice } from 'state/notices/actions';
+import { CONCIERGE_APPOINTMENT_CREATE, CONCIERGE_APPOINTMENT_RESCHEDULE } from 'state/action-types';
+import {
+	CONCIERGE_STATUS_BOOKED,
+	CONCIERGE_STATUS_BOOKING,
+	CONCIERGE_STATUS_BOOKING_ERROR,
+	CONCIERGE_ERROR_NO_AVAILABLE_STAFF,
+} from 'me/concierge/constants';
+import fromApi from './from-api';
+import toApi from './to-api';
+import { recordTracksEvent, withAnalytics } from 'state/analytics/actions';
+
+export const bookConciergeAppointment = action => {
+	return [
+		updateConciergeBookingStatus( CONCIERGE_STATUS_BOOKING ),
+		http(
+			{
+				method: 'POST',
+				path: `/concierge/schedules/${ action.scheduleId }/appointments`,
+				apiNamespace: 'wpcom/v2',
+				body: toApi( action ),
```

Comment:
wonderful use of `toApi()` to demarcate internal data representations from those the API expects!

Reply:
:bowtie: